src/helpers.ts

```diff
@@ -4,7 +4,7 @@ const globalObj = typeof window === 'undefined' ? global : window
 const TEXT_NODE = 3
 
 // Currently this fn only supports jest timers, but it could support other test runners in the future.
-function runWithRealTimers(callback) {
+function runWithRealTimers<T>(callback: () => T) {
```

Comment:
```suggestion
function runWithRealTimers(callback: () => unknown) {
```

Same as above: If we don't use the generic parameter, we can remove it.

Reply:
Maybe I miss something here. With this approach, if I call `runWithRealTimers` then I should cast the return value of the callback 

https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAJ3AVTAazHA7mAdRigAsAlAUwEMAbAFRgFsLkBnACglpoCMqJMALkTsAlIgC8APkThseMOIDeAKERrEEBKyibufAZSghkYAGq0QFSXpq9+mMSo0bkFY6dv3D7k+csUKgC+zipaYDoovqYWNFZo8vg2qGAYiYTE5NT0TCwcYpIyAOQkFHZwReJUrIg6yDBgAOahoJCwCCjgAOIUYCwwEESklLQMzGwAPHRSnPoOwgXSiHTKahrhkVx2BphGfrFWNlvejqLOiOoXbh5Ixzt7MQHBoRu61-sByd29-YOZIzlxvlxEsSmUaBUzkA

Reply:
```suggestion
function runWithRealTimers<T>(callback: () => T): T {
```
to make it obvious we're using the parameter.